File: LeetCode/Python/Easy/3062-Winner-of-the-Linked-List-Game/solution.py
class Solution:
    def gameResult(self, head: Optional[ListNode]) -> str:
        result="Tie"

        even_score=0

        odd_score=0

        current=head

        while current and current.next:
            if current.val > current.next.val:
                even_score+=1
            elif current.next.val > current.val:
                odd_score+=1

            current=current.next.next

        if odd_score!=even_score:
            result="Odd" if odd_score > even_score else "Even"

        return result

File: LeetCode/Rust/Easy/3062-Winner-of-the-Linked-List-Game/solution.rs
impl Solution
{
    pub fn game_result(head: Option<Box<ListNode>>) -> String
    {
        let mut result: String=String::from("Tie");

        let mut current: &Option<Box<ListNode>>=&head;

        let mut even_score: i32=0;

        let mut odd_score: i32=0;

        while let Some(node)=current
        {
            if let Some(next_node)=&node.next
            {
                if node.val > next_node.val
                {
                    even_score+=1;
                }
                else if next_node.val > node.val
                {
                    odd_score+=1;
                }

                current=&next_node.next;
            }
            else
            {
                break;
            }
        }

        if even_score!=odd_score
        {
            if even_score > odd_score
            {
                result="Even".to_string();
            }
            else
            {
                result="Odd".to_string();
            }
        }

        result
    }
}